core/src/main/java/com/linecorp/armeria/internal/common/AbstractKeepAliveHandler.java

```diff
@@ -181,6 +189,26 @@ public final void onReadOrWrite() {
             lastConnectionIdleTime = System.nanoTime();
         }
 
+        if (!isServer && !isActive) {
+            isActive = true;
+
+            final SessionProtocol protocol = HttpSession.get(channel).protocol();
+            final InetSocketAddress localAddr = ChannelUtil.localAddress(channel);
+            final InetSocketAddress remoteAddr = ChannelUtil.remoteAddress(channel);
```

Comment:
These values are immutable so it can cache the value in the member fields.
Should we set the values if `connectionPoolListener != ConnectionPoolListener.noop()`? If the values are null, we can assume there is no listeners.